plugins/sharding/logical_count.rb

```diff
@@ -56,179 +41,134 @@ def cache_key(input)
         key
       end
 
-      class ShardCountContext
-        attr_reader :shard
-        attr_reader :cover_type
-        attr_reader :range_index
-        attr_accessor :table
-        def initialize(shard, cover_type, range_index)
-          @shard = shard
-          @cover_type = cover_type
-          @range_index = range_index
-          @table = shard.table
+      class ShardCountContext < StreamExecuteContext
+        attr_reader :order
```

Comment:
It seems that we should be defined in `StreamExecuteContext`.

How about passing `order` to `StreamExecuteContext#initialize`?

Reply:
Please let me work on this in another PR because it also affects to `logical_range_filter`.

Reply:
#1347

Reply:
#1347 is merged